core/state_processor.go

```diff
@@ -86,6 +86,10 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 // for the transaction, gas used and an error if the transaction failed,
 // indicating the block was invalid.
 func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, error) {
+	if !config.IsYoloV1(header.Number) && tx.Type() != types.LegacyTxId {
```

Comment:
Curious is this is the appropriate place to make this consensus check?

Reply:
I think we should perform this check in the signer instead.